src/LayerPlanBuffer.cpp

```diff
@@ -38,41 +38,51 @@ void LayerPlanBuffer::insertPreheatCommand(ExtruderPlan& extruder_plan_before, d
         if (acc_time > time_after_extruder_plan_start)
         {
             const double time_before_path_end = acc_time - time_after_extruder_plan_start;
-            extruder_plan_before.insertCommand(path_idx, extruder, temp, false, time_this_path - time_before_path_end);
+            bool wait = false;
+            extruder_plan_before.insertCommand(path_idx, extruder, temp, wait, time_this_path - time_before_path_end);
             return;
         }
     }
-    extruder_plan_before.insertCommand(0, extruder, temp, false); // insert at start of extruder plan if time_after_extruder_plan_start > extruder_plan.time
+    bool wait = false;
+    unsigned int path_idx = 0;
+    extruder_plan_before.insertCommand(path_idx, extruder, temp, wait); // insert at start of extruder plan if time_after_extruder_plan_start > extruder_plan.time
 }
 
 Preheat::WarmUpResult LayerPlanBuffer::timeBeforeExtruderPlanToInsert(std::vector<ExtruderPlan*>& extruder_plans, unsigned int extruder_plan_idx)
 {
     ExtruderPlan& extruder_plan = *extruder_plans[extruder_plan_idx];
     int extruder = extruder_plan.extruder;
-    double required_temp = extruder_plan.required_temp;
-    
+    double initial_print_temp = extruder_plan.initial_printing_temperature;
+
     double in_between_time = 0.0;
     for (unsigned int extruder_plan_before_idx = extruder_plan_idx - 1; int(extruder_plan_before_idx) >= 0; extruder_plan_before_idx--)
     { // find a previous extruder plan where the same extruder is used to see what time this extruder wasn't used
-        ExtruderPlan& extruder_plan = *extruder_plans[extruder_plan_before_idx];
-        if (extruder_plan.extruder == extruder)
+        ExtruderPlan& extruder_plan_before = *extruder_plans[extruder_plan_before_idx];
+        if (extruder_plan_before.extruder == extruder)
         {
-            Preheat::WarmUpResult warm_up = preheat_config.timeBeforeEndToInsertPreheatCommand_coolDownWarmUp(in_between_time, extruder, required_temp);
+            double temp_before = preheat_config.getFinalPrintTemp(extruder);
+            if (temp_before == 0)
```

Comment:
0 means: 'don't use the final_print_temp feature'. This is different than just putting the final print temperature to the same value as the print temperature for the initial layer and for when flow dependent temperatures are used.